source/transfer_list.rst

```diff
@@ -657,5 +658,86 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+.. _arm_entries:
+
+Arm specific entries
+^^^^^^^^^^^^^^^^^^^^
+
+The following entry types are defined for Arm projects,
+including TF-A, OP-TEE and FF-A:
+
+**OP-TEE pageable part address entry layout (XFERLIST_OPTEE_PAGEABLE_PART_ADDR)**
+
+This entry type holds the address of OP-TEE pageable part which is described in
+[OPTEECore]_.
+This address (of type 'uint64_t') is used when OPTEED (OP-TEE Dispatcher)
+is the Secure Payload Dispatcher, indicating where to load the pageable image of
+the OP-TEE OS.
+
+.. _tab_optee_pageable_part_address:
+.. list-table:: OP-TEE pageable part address type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x100**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size (in bytes) of the address of OP-TEE pageable part which must be set to **8**.
+
+   * - pp_addr
+     - data_size
+     - hdr_size
+     - Holds the address of OP-TEE pageable part
+
+**SPMC manifest entry layout (XFERLIST_SPMC_MANIFEST)**
```

Comment:
I think the way this is used in TF-A is more general than this. TF-A defines the concept of a FW_CONFIG. In the context of SPMC configs, TOS_FW_CONFIG is overloaded as the SPMC manifest. However, in other configs, this translates into a BL specific config using the DT format. So I think this should be something like XFERLIST_TFA_FW_CONFIG, and explain that the format of the fw_config is statically known by the receiver. For example, it may be an SPMC manifest or a receiver specific configuration image.

Reply:
But seems that `FW_CONFIG` is too general and actually it can mean anything. For example, even FDT and HOB are some kind of `FW_CONFIG`. This is not good for a TE's definition.

Reply:
> But seems that `FW_CONFIG` is too general and actually it can mean anything. For example, even FDT and HOB are some kind of `FW_CONFIG`. This is not good for a TE's definition. I think better to define different TEs in the future for different receivers and purposes instead of just a general `FW_CONFIG`.

Agree, I also have same thoughts !

Reply:
Yes, FW_CONFIG is a general concept but that is how the code is written today so my suggestion is to just to align the spec with the code. It's true the definition of what's in the FW_CONFIG is then outside the spec but it at least provides a solution for passing all types of FW_CONFIG, not just this specific SPMC manifest case.
We can discuss changing TF-A to define FW_CONFIGs with specific content for each case (instead of DT) but I don't think this PR is the best place for that conversation. That would need a separate PR anyway.

Reply:
Once a TE entry is defined it cannot be changed anymore due to backward compatibility, I think now it is the time to define the new TEs with their real purposes instead of something general - Even right now the code is still using the old concepts.
If we still define `FW_CONFIG` here, we will have to deprecate it in a near future.

Reply:
I think now the question is whether we should keep the spec to reflect to the current code or the spec should move ahead from the code? Hmmm... I prefer to the latter.

Reply:
It's hard to change an existing TE but it's easy to create new ones and use them instead. However, I agree it would be better to align with where we want to go rather than change later. I personally don't see a strong need to change from the existing FW_CONFIG - it makes it easy to extend without requiring spec changes. I will talk to @manish-pandey-arm any other to at least get an agreed Arm position.

Reply:
> Dan are you suggesting that we might add new members to this struct later, increasing its size?

No, we're just discussing whether to overload the meaning of the data in this TE or if there's a risk of us changing the meaning in future.

I discussed this more with @manish-pandey-arm. Manish wants to explore passing more specific TEs for some BLs/configs to avoid the overhead of DT parsing in some images. However, we think are some configs/platforms where it will always be necessary to pass some kind of BL-specific config in device tree format separate to XFERLIST_FDT. A generic fw_config TE could be a solution to that, but that does not need to be addressed in this PR. While such a TE could potentially also be used for this SPMC manifest, I'm now happy for a separate TE to be defined for this. It might help with parsing in the producer, especially if additional BL-specific config data needs to be passed later.

I have a few more questions/comments:

- According to the FF-A spec, the format of the SPMC manifest is implementation defined. I believe you're intending to use this in TF-A where the DT format is used, right? In which case, the DT format should be specified here and the name should specify DT to enable other SPMC formats, e.g. XFERLIST_DT_SPMC_MANIFEST.

- The FF-A spec defines it's own handoff protocol (called FF-A boot protocol), which competes with this spec. Is it a conscious decision to use this spec instead of the FF-A boot protocol? I'd say that's good idea anyway as this spec is likely to be more dynamic that the FF-A spec.

- If this was an FF-A TE, then I would say it should be in a separate (e.g. Arm architecture) range of TEs. However, if my above comment is correct, this is really still a TF-A TE since TF-A chooses the SPMC manifest format.

Reply:
@danh-arm Seems to be a reasonable solution.
I will rename the TE tag to `XFERLIST_DT_SPMC_MANIFEST` and the group name to `TF-A related`.
Link to 'https://trustedfirmware-a.readthedocs.io/en/latest/components/secure-partition-manager.html#spmc-manifest' instead of FF-A spec.

Reply:
Done.